### PR TITLE
util:  fix can use invisible index after admin check table

### LINKED ISSUE #26217: can use invisible index after admin check table
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

``` SQL
mysql> create table te(a int, unique key(a) invisible);
Query OK, 0 rows affected (0.00 sec)

mysql> desc select * from te where a= 3;
+-------------------------+----------+-----------+---------------+--------------------------------+
| id                      | estRows  | task      | access object | operator info                  |
+-------------------------+----------+-----------+---------------+--------------------------------+
| TableReader_7           | 1.00     | root      |               | data:Selection_6               |
| └─Selection_6           | 1.00     | cop[tikv] |               | eq(test.te.a, 3)               |
|   └─TableFullScan_5     | 10000.00 | cop[tikv] | table:te      | keep order:false, stats:pseudo |
+-------------------------+----------+-----------+---------------+--------------------------------+
3 rows in set (0.00 sec)

mysql> admin check table te;
Query OK, 0 rows affected (0.01 sec)

mysql> desc select * from te where a= 3;
+-------------+---------+------+----------------------+---------------+
| id          | estRows | task | access object        | operator info |
+-------------+---------+------+----------------------+---------------+
| Point_Get_5 | 1.00    | root | table:te, index:a(a) |               |
+-------------+---------+------+----------------------+---------------+
1 row in set (0.00 sec)
```
### 2. What did you expect to see? (Required)
Can't use the invisible index.
### 3. What did you see instead (Required)
Use the invisible index
### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by wjhuang2016**
### What problem does this PR solve?

Issue Number: close #26217 

Problem Summary:
When admin check table, OptimizerUseInvisibleIndexes is set to true, but isn't reset later.

### What is changed and how it works?

Reset OptimizerUseInvisibleIndexes

How it Works:

### Check List 

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

- No release note

### CODE REVIEW

**Review by winoros — APPROVED** [review]

File: util/admin/admin.go [hunk 1]
```diff
@@ -316,6 +316,9 @@ const (
 func CheckIndicesCount(ctx sessionctx.Context, dbName, tableName string, indices []string) (byte, int, error) {
 	// Here we need check all indexes, includes invisible index
 	ctx.GetSessionVars().OptimizerUseInvisibleIndexes = true
+	defer func() {
+		ctx.GetSessionVars().OptimizerUseInvisibleIndexes = false
```
Comment (zimulala) [inline util/admin/admin.go:320]: Set it to the original value is better？

**Review by AilinKid — APPROVED** [review]
LGTM

File: util/admin/admin.go (hunk 1 above)
Comment (wjhuang2016) [reply util/admin/admin.go:320]: It's ok to set it to false here. Because we use the pattern:
``` SQL
set it to true
lalala
reset it
```

Comment (zimulala) [reply util/admin/admin.go:320]: Got it.

**Pull request merged**